crates/datasources/src/object_store/http.rs

```diff
@@ -65,6 +65,11 @@ impl ObjStoreAccess for HttpStoreAccess {
         store: &Arc<dyn ObjectStore>,
         pattern: &str,
     ) -> Result<Vec<ObjectMeta>> {
+        if pattern.contains('*') {
+            return Err(ObjectStoreSourceError::Static(
+                "Globbing not supported for HTTP store",
+            ));
+        }
```

Comment:
I don't know if we want to return an error here. What if there's a URL with `*` in it? Do we expect each URL to have %-encoded chars?

Also, there are more chars that can say it's a glob, for eg., `?`.

Reply:
> I don't know if we want to return an error here. What if there's a URL with * in it? Do we expect each URL to have %-encoded chars?

see my initial comment. I'd assume any actual `*` would be url encoded.

Reply:
@vrongmeal I updated the logic to better handle files with `*` in them.